File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if str(x) == str(x)[::-1]:
            return True
        else:
            return False


solution = Solution()


print(solution.isPalindrome(121)) 
print(solution.isPalindrome(123)) 

File: 9-palindrome-number/palindrome-number.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x < 0 {
            return false;
        }

        let mut original = x;
        let mut reversed = 0;

        while original != 0 {
            let digit = original % 10;
            reversed = reversed * 10 + digit;
            original /= 10;
        }

        x == reversed
    }
}
